### PR TITLE
Support enum member as type

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/1922

### CODE REVIEW

File: packages/typespec-ts/test/unit/clientFactoryGenerator.spec.ts [hunk 1]
```diff
@@ -302,44 +305,19 @@ describe("Client Factory generation", () => {
...
-        
-        return client;
-    }
-    `
-      );
+      `,
+          false,
+          false
+        );
+        assert.fail("Should throw diagnostic errors");
```
Comment (qiaozha) [inline packages/typespec-ts/test/unit/clientFactoryGenerator.spec.ts:312]: Is it come from anywhere that we should throe diagnostic errors ?

File: packages/typespec-ts/test/unit/modelsGenerator.spec.ts [hunk 2]
```diff
@@ -712,14 +712,110 @@ describe("Input/output model type", () => {
...
+            assertEqualContent(
+              outputModelFile?.content!,
+              `
+            export interface COutput extends BOutput {
+              a: 1.1;
+            }
+    
+            export interface BOutput {
+              /** Possible values: 1.1, 2.2 */
+              a: string;
```
Comment (qiaozha) [inline packages/typespec-ts/test/unit/modelsGenerator.spec.ts:813]: Should this be number in this case ?

File: packages/typespec-ts/test/integration/modelInheritance.spec.ts [hunk 3]
```diff
@@ -14,7 +15,7 @@ describe("ModelsInheritance Rest Client", () => {
     });
   });
 
-  const validBody = { name: "abc", age: 32, smart: true };
+  const validBody: Siamese = { name: "abc", age: 32, smart: true };
```
Comment (qiaozha) [inline packages/typespec-ts/test/integration/modelInheritance.spec.ts:18]: not sure if we shoud add this, as we are recommending our customers to not use type annotation now.

Comment (MaryGao) [reply packages/typespec-ts/test/integration/modelInheritance.spec.ts:18]: I think this is only for inheritance test case, we could add the interface name explicitly for type checking.

File: packages/typespec-ts/test/unit/modelsGenerator.spec.ts (hunk 2 above)
Comment (MaryGao) [reply packages/typespec-ts/test/unit/modelsGenerator.spec.ts:813]: number type is not allowed for tsp compiler so we would always fail in this case and in UT we have no way to catch this error.

File: packages/typespec-ts/test/unit/util/emitUtil.ts [hunk 4]
```diff
@@ -108,6 +113,11 @@ export async function emitClientFactoryFromCadl(
   if (clients && clients[0]) {
     apiVersionInfo = transformApiVersionInfo(clients[0], dpgContext, urlInfo);
   }
+  if (isEmptyDiagnostic) {
+    expectDiagnosticEmpty(dpgContext.program.diagnostics);
+  } else {
+    throw dpgContext.program.diagnostics;
```
Comment (MaryGao) [inline packages/typespec-ts/test/unit/util/emitUtil.ts:119]: here to throw error.

File: packages/typespec-ts/test/unit/clientFactoryGenerator.spec.ts (hunk 1 above)
Comment (MaryGao) [reply packages/typespec-ts/test/unit/clientFactoryGenerator.spec.ts:312]: https://github.com/Azure/autorest.typescript/pull/1924/files/3ab05f96704bfa7e4ff074e474e63808e1e23963#r1265061076

File: packages/typespec-ts/test/unit/modelsGenerator.spec.ts [hunk 5]
```diff
@@ -239,28 +239,52 @@ describe("Input/output model type", () => {
...
+          cadlType,
+          typeScriptType,
+          {
+            additionalCadlDefinition: cadlTypeDefinition
+          },
+          true
+        );
+      });
+
+      it("with enum value is xx.xx", async () => {
```
Comment (MaryGao) [inline packages/typespec-ts/test/unit/modelsGenerator.spec.ts:266]: Add one UT

**Review by qiaozha — APPROVED** [review]

**Pull request merged**